File: app/src/main/java/afm/anime/AnimeType.java
package afm.anime;

import javax.annotation.Nonnull;

public enum AnimeType {
	TV,
	OVA,
	MOVIE,
	SPECIAL,
	ONA,
	MUSIC,  // MAL but not the API i got this from
	UNKNOWN,
	;

	public static @Nonnull AnimeType valueOfOrUnknown(@Nonnull String name) {
		for (AnimeType type : values()) {
			if (type.name().equalsIgnoreCase(name))
				return type;
		}
		return UNKNOWN;
	}
}

File: app/src/main/java/afm/anime/AnimeType.kt
package afm.anime

enum class AnimeType {
    TV,
    OVA,
    MOVIE,
    SPECIAL,
    ONA,
    MUSIC,  // MAL but not the API i got this from
    UNKNOWN,
    ;

    companion object {
        @JvmStatic
        fun valueOfOrUnknown(name: String): AnimeType =
            values().firstOrNull {
                it.name.equals(name, ignoreCase = true)
            } ?: UNKNOWN
    }
}
